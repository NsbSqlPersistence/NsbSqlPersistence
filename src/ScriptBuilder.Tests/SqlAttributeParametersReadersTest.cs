using System.Collections.Generic;
using NUnit.Framework;
#if NET452
using ObjectApproval;
#endif

[TestFixture]
public class SqlAttributeParametersReadersTest
{

    [Test]
    public void Minimal()
    {
        var result = SettingsAttributeReader.ReadFromAttribute(
            new CustomAttributeMock(
                new Dictionary<string, object>
                {
                    {
                        //At least one is required
                        "MsSqlServerScripts", true
                    }
                }));

        Assert.IsNotNull(result);
#if NET452
        ObjectApprover.VerifyWithJson(result);
#endif
    }

    [Test]
    public void Defaults()
    {
        var result = SettingsAttributeReader.ReadFromAttribute(null);
        Assert.IsNotNull(result);
#if NET452
        ObjectApprover.VerifyWithJson(result);
#endif
    }

    [Test]
    public void NonDefaults()
    {
        var result = SettingsAttributeReader.ReadFromAttribute(
            new CustomAttributeMock(
                new Dictionary<string, object>
                {
                    {
                        "ScriptPromotionPath", @"D:\scripts"
                    },
                    {
                        "MsSqlServerScripts", true
                    },
                    {
                        "MySqlScripts", true
                    },
                    {
                        "OracleScripts", true
                    },
                    {
                        "ProduceSagaScripts", false
                    },
                    {
                        "ProduceTimeoutScripts", false
                    },
                    {
                        "ProduceSubscriptionScripts", false
                    },
                    {
                        "ProduceOutboxScripts", false
                    }
                }));
        Assert.IsNotNull(result);
#if NET452
        ObjectApprover.VerifyWithJson(result);
#endif
    }


}