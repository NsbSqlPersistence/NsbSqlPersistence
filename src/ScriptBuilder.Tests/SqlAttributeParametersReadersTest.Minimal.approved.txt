{
  "BuildDialects": [
    "MsSqlServer"
  ],
  "ScriptPromotionPath": null,
  "ProduceOutboxScripts": true,
  "ProduceSubscriptionScripts": true,
  "ProduceTimeoutScripts": true,
  "ProduceSagaScripts": true
}